### PR TITLE
Resolved OSX and Move-calculation warnings

### PR DESCRIPTION
**Pull request by RoiEXLab**
The only warnings left are because of FileSystemAccessStrategy.remove() and SoftJEditorPane.
If someone told me on what to use instead / how to edit the remove() method I would do this in another pull request...
I'm not sure if these changes work like expected - Someone owning a mac would need to check that part, but I'm mostly concerned about the move calculation part. Since some methods don't seem to be called, I cant check if my changes work - seems to work fine until now.

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/609)

### CODE REVIEW

File: src/games/strategy/engine/lobby/client/ui/MacLobbyWrapper.java [hunk 1]
```diff
@@ -13,10 +14,9 @@
   // i think the java validator triggers this
   public static void registerMacShutdownHandler(final LobbyFrame frame) {
     // new Application();
-    final Application application = Application.getApplication();
-    application.addApplicationListener(new ApplicationAdapter() {
+    Application.getApplication().setQuitHandler(new QuitHandler(){
       @Override
-      public void handleQuit(final ApplicationEvent event) {
+      public void handleQuitRequestWith(QuitEvent arg0, QuitResponse arg1) {
```
Comment (DanVanAtta) [inline src/games/strategy/engine/lobby/client/ui/MacLobbyWrapper.java]: Please rename parameters `arg0` and `arg1`
I recommend 'quitEvent' and 'quitResponse'

File: src/games/strategy/triplea/ai/proAI/ProBidAI.java [hunk 2]
```diff
@@ -1765,7 +1765,11 @@ private static float determineEnemyBlitzStrength(final Territory blitzHere, fina
             continue;
           }
           for (final Unit u : neighbor.getUnits()) {
-            if (unitCondition.match(u) && MoveValidator.hasEnoughMovement(u, dist)) {
+            Route r1 = new Route();
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ai/proAI/ProBidAI.java]: pls rename `r1` to `route` (or similar)

File: src/games/strategy/triplea/ai/proAI/ProTechAI.java [hunk 3]
```diff
@@ -429,7 +429,11 @@ private static float determineEnemyBlitzStrength(final Territory blitzHere, fina
             continue;
           }
           for (final Unit u : neighbor.getUnits()) {
-            if (unitCondition.match(u) && MoveValidator.hasEnoughMovement(u, dist)) {
+            Route r1 = new Route();
+            for(Route r : routes){
+              r1 = Route.join(r1, r);
+            }
+            if (unitCondition.match(u) && Matches.UnitHasEnoughMovementForRoute(r1).match(u)) {
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ai/proAI/ProTechAI.java]: DRY (don't repeat yourself)  violation.
Add a `Matches.UnitHasEnoughMovementForRoute(Collection<Route> routes )` method that does the join for you.

File: src/games/strategy/triplea/ai/proAI/ProBidAI.java [hunk 4]
```diff
@@ -1837,10 +1841,10 @@ private static float determineEnemyBlitzStrength(final Territory blitzHere, fina
       }
     }
     for (final Unit u : unitDistance.keySet()) {
-      if (lz != null && MoveValidator.hasEnoughMovement(u, unitDistance.getInt(u) + distance.getInt(lz))) {
+      if (lz != null && Matches.UnitHasEnoughMovementForRoute(new Route(checked)).match(u)) {
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ai/proAI/ProBidAI.java]: `new Route(checked)` call into the Matches function is redundant. Pls Overload the method to take a  `List<Territory>` as well

**Comment by RoiEXLab** [comment]
Ok not everything could be implemented like you told be, because Java does not differ between different generic types of the same type e.g. List<Territories> and List<Routes> do not differ -> error

I have 2 questions for future pull requests...

- What are Mock elements? Mentioned in TestDelegateBridge to be used instead.
- What should be used instead of SoftJEditorPane?

These classes are the last ones creating warnings...

**Comment by DanVanAtta** [comment]
### mock
Putting the answer here: https://github.com/triplea-game/triplea/wiki/Code-Standards
The page is a work in progress, but better to have it there rather than get lost in closed PRs 

### SoftJEditorPane
We would perhaps have to build a replacement. Hopefully we can iterate to a simpler design where we just wont' need it, or at least simplify it so it is more trivial.

**Pull request merged**